src/System.Reflection.Metadata.ILDasm/src/Decoder/ILDecoder.cs

```diff
@@ -0,0 +1,582 @@
+                        break;
+                    case OperandType.ShortInlineR:
+                        instruction = new ILSingleInstruction(opCode, ilReader.ReadSingle(), -1, expectedSize + 4);
+                        break;
+                    case OperandType.InlineNone:
+                        instruction = new ILInstructionWithNoValue(opCode, expectedSize);
+                        break;
+                    case OperandType.ShortInlineVar:
+                        byte token = ilReader.ReadByte();
+                        instruction = new ILInt16VariableInstruction(opCode, GetVariableName(opCode, token, _methodDefinition), token, expectedSize + 1);
+                        break;
+                    case OperandType.InlineVar:
+                        shortOperand = ilReader.ReadUInt16();
+                        instruction = new ILVariableInstruction(opCode, GetVariableName(opCode, shortOperand, _methodDefinition), shortOperand, expectedSize + 2);
+                        break;
+                    case OperandType.InlineSig:
+                        intOperand = ilReader.ReadInt32();
+                        instruction = new ILStringInstruction(opCode, GetSignature(mdReader, intOperand, provider), intOperand, expectedSize + 4);
+                        break;
+                    default:
+                        break;
+                }
+                ilOffset += instruction.Size;
+                yield return instruction;
+            }
+        }
+
+        internal static ILLocal[] DecodeLocalSignature(MethodBodyBlock methodBody, MetadataReader mdReader, ILTypeProvider provider)
+        {
+            if (methodBody.LocalSignature.IsNil)
+            {
+                return new ILLocal[0];
+            }
+            var localTypes = SignatureDecoder.DecodeLocalSignature(methodBody.LocalSignature, provider);
+            ILLocal[] locals = new ILLocal[localTypes.Count()];
+            for (int i = 0; i < localTypes.Length; i++)
+            {
+                string name = "V_" + i;
+                locals[i] = new ILLocal(name, localTypes[i].ToString());
+            }
+            return locals;
+        }
+
+        internal static ILParameter[] DecodeParameters(MethodSignature<ILType> signature, ParameterHandleCollection parameters, ref Readers readers)
+        {
+            var types = signature.ParameterTypes;
+            int requiredCount = Math.Min(signature.RequiredParameterCount, types.Length);
+            if (requiredCount == 0)
+            {
+                return new ILParameter[0];
+            }
+            ILParameter[] result = new ILParameter[requiredCount];
+            for (int i = 0; i < requiredCount; i++)
+            {
+                var parameter = readers.MdReader.GetParameter(parameters.ElementAt(i));
+                result[i] = ILParameter.Create(parameter, ref readers, types[i].ToString());
```

Comment:
Should there be a Create overload that does not require calling ToString on the third parameter?